### PR TITLE
Setup firebase client

### PR DESCRIPTION
**Pull request by alexknipfer**
Firebase client is functional, you can:

- Sign in with email and password
- Get auth status
- Get current users profile

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,12 @@
   "dependencies": {
     "@stencil/core": "^0.12.4",
     "@stencil/router": "^0.2.6",
+    "firebase": "^5.5.1",
+    "inversify": "^4.13.0",
+    "inversify-inject-decorators": "^3.1.0",
     "ionicons": "^4.4.3",
+    "quill": "^1.3.6",
```
Comment (knipferrc) [inline package.json]: Shouldnt have quill in here

File: src/components/app-root/app-root.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import '../../config/InversifyBinder'
```
Comment (knipferrc) [inline src/components/app-root/app-root.tsx]: Out of curiosity, what does this do?

File: src/services/FirebaseClient/FirebaseClient.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import firebase from 'firebase/app'
```
Comment (knipferrc) [inline src/services/FirebaseClient/FirebaseClient.ts]: Is this file just acting as a contract to the impl file?

File: src/components/app-root/app-root.tsx (hunk 2 above)
Comment (alexknipfer) [reply src/components/app-root/app-root.tsx]: Binds our classes at startup.

File: src/services/FirebaseClient/FirebaseClient.ts (hunk 3 above)
Comment (alexknipfer) [reply src/services/FirebaseClient/FirebaseClient.ts]: Just an interface for the class, I guess if you want to call it a contract.

**Pull request merged**